jerry-core/include/jerryscript-compiler.h

```diff
@@ -31,7 +31,7 @@ extern "C"
  * Compiler-specific macros relevant for GCC.
  */
 #define JERRY_ATTR_ALIGNED(ALIGNMENT) __attribute__((aligned(ALIGNMENT)))
-#define JERRY_ATTR_ALWAYS_INLINE __attribute__((always_inline))
+#define JERRY_ATTR_ALWAYS_INLINE inline __attribute__((always_inline))
```

Comment:
That's not exactly as suggested. I've suggested to keep the GCC version of `JERRY_ATTR_ALWAYS_INLINE` unchanged.

Reply:
then read the code carefully, you are suggesting invalid.things

Reply:
> That's not exactly as suggested. I've suggested to keep the GCC version of `JERRY_ATTR_ALWAYS_INLINE` unchanged.

inline and __forceinline can not co-exist.  read the title first.

Reply:
charming

Reply:
Frankly, this is a C99 project primarily targeting resource-constrained embedded systems. Windows is not a resource-constrained platform and more importantly, MSVC is not C99 conforming. I'm not really convinced to replace even the standard C `inline` keyword for the sake of MSVC.

This PR aims at removing the need for amalgamated builds in case of MSVC. Then it should do that, and only that. According to what has been revealed so far, the only thing necessary to achieve that is to add `extern` in front of `inline` in same cases. No new macros, no macro re-purposing,  no macro placement reordering.

If `__forceinline` is not a modifier, like `__attribute__(())` or `__declspec()`, but a replacement for the `inline` keyword then we should not consider adding it to the mix, at least for now. Then `JERRY_ATTR_ALWAYS_INLINE` will remain empty for MSVC. Should not be a big deal, the project will compile even if some functions are "simple" inlines, not forced ones.

Reply:
@akosthekiss done